test/search_query_test.rb

```diff
@@ -0,0 +1,28 @@
+require 'test_helper'
+
+class SearchQueryTest < Minitest::Test
+  let(:connection_options) { { :token => 'dummy' } }
+  let(:connection) { Papertrail::Connection.new(connection_options) }
+  let(:search_query) { Papertrail::SearchQuery.new }
+  let(:get_response) { OpenStruct.new(:body => { 'max_id' => 122, 'events' => [] }) }
+  let(:api_url) { Papertrail::SearchQuery.api_url }
+  let(:default_initial_params) { { limit: Papertrail::SearchQuery.initial_search_limit } }
+
+  def test_sets_min_id_for_subsequent_requests_with_max_id
+    # Also testing that we set the initial_search_limit only for the first request
+    connection.expects(:get).with(api_url, default_initial_params).returns(get_response)
+    search_query.search_results(connection)
+    target = {
+      :min_id => 122,
+      limit: Papertrail::SearchQuery.subsequent_search_limits
+    }
+    connection.expects(:get).with(api_url, target).returns(get_response)
+    search_query.search_results(connection)
+  end
+
+  def test_sets_limit_from_options
+    opts = { limit: 2 }
+    connection.expects(:get).with(api_url, opts).returns(get_response)
+    search_query.search_results(connection, nil, opts)
```

Comment:
It uses whatever options it's passed - if a subsequent request doesn't pass it options, it will use the `default_request_limit`. So another test wouldn't be useful, from my perspective